test/virtual_dataset.jl

```diff
@@ -0,0 +1,38 @@
+using Test, HDF5
+
+dirname = mktempdir()
+
+filename = joinpath(dirname, "main.hdf5")
+
+f = h5open(filename, "w")
+
+sub0 = joinpath(dirname, "sub-0.hdf5")
+f0 = h5open(sub0, "w")
+f0["x"] = fill(1.0, 3)
+close(f0)
+
+sub1 = joinpath(dirname, "sub-1.hdf5")
+f1 = h5open(sub1, "w")
+f1["x"] = fill(2.0, 3)
+close(f1)
+
+srcspace = dataspace((3,))
+vspace = dataspace((3, 2); max_dims=(3, -1))
+HDF5.select_hyperslab!(vspace, (1:3, HDF5.BlockRange(1; count=-1)))
+
+d = create_dataset(
+    f,
+    "x",
+    datatype(Float64),
+    vspace;
+    virtual=[HDF5.VirtualMapping(vspace, "./sub-%0b.hdf5", "x", srcspace)]
```

Comment:
I'm getting the following error:

Is the directory name supposed to be  "./sub-01.hdf5" ? 

```julia
julia> d = create_dataset(
           f,
           "x",
           datatype(Float64),
           vspace;
           virtual=[HDF5.VirtualMapping(vspace, "./sub-%0b.hdf5", "x", srcspace)]
       )
ERROR: HDF5.API.H5Error: Error setting virtual
libhdf5 Stacktrace:
 [1] H5D_virtual_parse_source_name: Invalid arguments to routine/Bad value
     invalid format specifier
  ⋮
Stacktrace:
  [1] macro expansion
    @ C:\Users\Mus\.julia\packages\HDF5\5P16B\src\api\error.jl:18 [inlined]
  [2] h5p_set_virtual(dcpl_id::HDF5.DatasetCreateProperties, vspace_id::HDF5.Dataspace, src_file_name::String, src_dset_name::String, src_space_id::HDF5.Dataspace)
    @ HDF5.API C:\Users\Mus\.julia\packages\HDF5\5P16B\src\api\functions.jl:3960
  [3] push!
    @ C:\Users\Mus\.julia\packages\HDF5\5P16B\src\virtual.jl:43 [inlined]
  [4] append!
    @ C:\Users\Mus\.julia\packages\HDF5\5P16B\src\virtual.jl:50 [inlined]
  [5] set_virtual!
    @ C:\Users\Mus\.julia\packages\HDF5\5P16B\src\properties.jl:507 [inlined]
  [6] class_setproperty!(#unused#::Type{HDF5.DatasetCreateProperties}, p::HDF5.DatasetCreateProperties, name::Symbol, val::Vector{HDF5.VirtualMapping})
    @ HDF5 C:\Users\Mus\.julia\packages\HDF5\5P16B\src\properties.jl:550
  [7] setproperty!
    @ C:\Users\Mus\.julia\packages\HDF5\5P16B\src\properties.jl:59 [inlined]
  [8] (::HDF5.var"#4#5"{Tuple{HDF5.DatasetCreateProperties, HDF5.DatasetTransferProperties, HDF5.DatasetAccessProperties}})(::Pair{Symbol, Vector{HDF5.VirtualMapping}})
    @ HDF5 C:\Users\Mus\.julia\packages\HDF5\5P16B\src\properties.jl:139
  [9] filter
    @ .\abstractdict.jl:472 [inlined]
 [10] #setproperties!#3
    @ C:\Users\Mus\.julia\packages\HDF5\5P16B\src\properties.jl:135 [inlined]
 [11] create_dataset(parent::HDF5.File, path::String, dtype::HDF5.Datatype, dspace::HDF5.Dataspace; dcpl::HDF5.DatasetCreateProperties, dxpl::HDF5.DatasetTransferProperties, dapl::HDF5.DatasetAccessProperties, pv::Base.Pairs{Symbol, Vector{HDF5.VirtualMapping}, Tuple{Symbol}, NamedTuple{(:virtual,), Tuple{Vector{HDF5.VirtualMapping}}}})
    @ HDF5 C:\Users\Mus\.julia\packages\HDF5\5P16B\src\datasets.jl:57
 [12] top-level scope
    @ REPL[20]:1
```

Reply:
```
julia> versioninfo()
Julia Version 1.8.2
Commit 36034abf26 (2022-09-29 15:21 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: 16 × AMD Ryzen 9 5900HS with Radeon Graphics
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, znver3)
  Threads: 16 on 16 virtual cores
Environment:
  JULIA_EDITOR = code.cmd
  JULIA_NUM_THREADS = auto
```

Reply:
Oops, we did not add virtual_dataset.jl to runtests.jl

```julia
julia> HDF5.API.SHORT_ERROR[] = false
false

julia> d = create_dataset(
           f,
           "x",
           datatype(Float64),
           vspace;
           virtual=[HDF5.VirtualMapping(vspace, "./sub-%0b.hdf5", "x", srcspace)]
       )
ERROR: HDF5.API.H5Error: Error setting virtual
libhdf5 Stacktrace:
 [1] H5D_virtual_parse_source_name: Invalid arguments to routine/Bad value
     invalid format specifier
 [2] H5Pset_virtual: Property lists/Unable to initialize object
     can't parse source file name

```

Reply:
Ah, it should be just `%b`: the docs are wrong, they never actually added support for the dimension specifier. I'll open a PR